### ISSUE TITLE
UnexpectedValueException with anonymous class inside function definition

### ISSUE THREAD

**Issue by kelunik** [FriendsOfPHP/PHP-CS-Fixer#2981]
PHP-CS-Fixer errors out for the following file.

The PHP version you are using:
```
PHP 7.0.22-0ubuntu0.17.04.1 (cli) (built: Aug  8 2017 22:03:30) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.22-0ubuntu0.17.04.1, Copyright (c) 1999-2017, by Zend Technologies
    with blackfire v1.17.3~linux-x64-non_zts70, https://blackfire.io, by SensioLabs
```

PHP CS Fixer version you are using:
```
PHP CS Fixer 2.4.0 Silver Gingerbread by Fabien Potencier and Dariusz Ruminski
```

The command you use to run the fixer (and the configuration file you are using if any).
```
vendor/bin/php-cs-fixer fix test.php -vv
```

A minimal sample of valid PHP code that is not fixed correctly (if applicable).
```php
<?php

namespace Aerys\Internal;

function buildVhost(Host $host, callable $bootLoader): Vhost {
    $application = $bootLoader(new class($applications) implements Bootable, ServerObserver {
        public function __invoke(Request $request, Response $response) {
            // nothing
        }
    });
}
```

Configuration: https://github.com/amphp/aerys/blob/f4e650f9f2ffb60dc1c8ff169129009a5db68078/.php_cs.dist

**Comment by julienfalque** [comment]
Thanks for your report. The bug was fixed in #2931, which was merged and should be released soon.

**Comment by kelunik** [comment]
Thanks, can confirm it's fixed with `php-cs-fixer@dev`.
